tests/test_timefuncs.py

```diff
@@ -1042,23 +1042,11 @@ def timedelta64_and_arithmatic_operator() -> None:
     s2 = pd.Series(data=pd.date_range("1/1/2021", "2/1/2021"))
     s3 = s2 - s1
     td = np.timedelta64(1, "M")
-    check(assert_type((s1 - td), TimestampSeries), np.timedelta64)
-    check(assert_type((s1 + td), pd.Series[Any]), np.timedelta64)
-    check(assert_type((s1 * td), TimedeltaSeries), np.timedelta64)
-    check(assert_type((s1 / td), pd.Series[float]), np.timedelta64)
-    check(
-        assert_type((s3 - td), TimedeltaSeries),
-        pd.Series,
-    )
-    check(
-        assert_type((s3 + td), TimedeltaSeries),
-        pd.Series,
-    )
-    check(
-        assert_type((s3 * td), TimedeltaSeries),
-        pd.Series,
-    )
-    check(
-        assert_type((s3 / td), pd.Series[float]),
-        pd.Series,
-    )
+    check(assert_type((s1 - td), TimestampSeries), pd.Series, pd.Timestamp)
+    check(assert_type((s1 + td), TimestampSeries), pd.Series, pd.Timestamp)
+    check(assert_type((s1 * td), TimedeltaSeries), pd.Series, pd.Timestamp)
+    check(assert_type((s1 / td), TimestampSeries), pd.Series, pd.Timestamp)
+    check(assert_type((s3 - td), TimedeltaSeries), pd.Series, pd.Timedelta)
+    check(assert_type((s3 + td), TimedeltaSeries), pd.Series, pd.Timedelta)
+    check(assert_type((s3 * td), TimedeltaSeries), pd.Series, pd.Timedelta)
```

Comment:
multiplying a `TimedeltaSeries` by a timedelta fails, so have to fix that

In `TimedeltaSeries`, put in a method for `__mul__()` that takes the two timedelta arguments, and returns `Never` from `typing_extensions`

Change the test to assert the type, but do not do the `check` part.

Reply:
And in `TimedeltaSeries ` should a `__trudiv__()` be added because it is showing errors

Reply:
and sir `__mul__()` it is still showing errors after adding  it.

Reply:
> Sir in `TimedeltaSeries` should a `__trudiv__()` be added because it is showing errors

Yes, you should add that.  It is valid to divide `TimedeltaSeries` by `Timedelta` or `np.timedelta64` to get a `Series[float]`, and to divde by a `float` to get `TimedeltaSeries`

Reply:
Sir just not able to fix this `error: Name "__mul__" already defined on line 1761  [no-redef]` could not find about it over internet also

Reply:
You probably have `__mul__()` defined in there twice, with different arguments.  You need to put a `@overload` before each `def __mul__()` statement

Reply:
Sir still mypy hows the same error

Reply:
Without seeing your code, I can't see what you are doing wrong.  You can push and it will fail and I will tell you what you did wrong.

Reply:
yes sir